### PR TITLE
Fix warnings for code used by blocks

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Compiler warnings.

### Proposed Changes

Fix warnings for code used by blocks.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/block_svg.js [hunk 1]
```diff
@@ -238,6 +238,19 @@ Blockly.BlockSvg.prototype.decompose;
  */
 Blockly.BlockSvg.prototype.compose;
 
+/**
+ * An optional boolean for determining whether or not the context menu is
+ * shown on a block.
+ * @type {?boolean}
+ */
+Blockly.BlockSvg.prototype.contextMenu;
```
Comment (alschmiedt) [inline core/block_svg.js]: Just curious why we need this here if it is already defined on the block?

Comment (samelhusseini) [reply core/block_svg.js]: Good point.

File: core/renderers/measurables/connections.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ Blockly.blockRendering.Connection = function(constants, connectionModel) {
       constants);
   this.connectionModel = connectionModel;
   this.shape = this.constants_.shapeFor(connectionModel);
-  this.isDynamicShape = !!this.shape.isDynamic;
+  this.isDynamicShape = !!this.shape['isDynamic'];
```
Comment (rachel-fenichel) [inline core/renderers/measurables/connections.js]: Why the change in style?

Comment (samelhusseini) [reply core/renderers/measurables/connections.js]: shapeFor is returning an object, we don't have any idea of its "shape" (no pun intended). 
When we use an interface for shapes this will go away.
